### PR TITLE
Transformations part 1

### PR DESCRIPTION
**Pull request by jnikula**
I got a bit stuck with adding tests for the transformations (see #33), but here's the preliminary work to clean stuff up, so I don't have to carry it locally.

### CODE REVIEW

File: hawkmoth/util/doccompat.py [hunk 1]
```diff
@@ -17,11 +17,13 @@
 # facilitate any kind of migration to Hawkmoth. The compat code could be turned
 # into a fairly simple plugin architecture, with some basic compat builtins, and
 # the users could still extend the compat features to fit their specific needs.
```
Comment (BrunoMSantos) [inline hawkmoth/util/doccompat.py:19]: Hmm, I was comparing with the RFC notes and I noticed this `FIXME` is still here. Is it not obsolete with this series?

**Review by BrunoMSantos — APPROVED** [review]
Looks good to me :)

File: hawkmoth/util/doccompat.py (hunk 1 above)
Comment (jnikula) [reply hawkmoth/util/doccompat.py:19]: Not yet. I have the implementation pretty much ready locally, but I had some issues testing it.

**Comment by jnikula** [comment]
Thanks again for your review!

File: hawkmoth/util/doccompat.py (hunk 1 above)
Comment (BrunoMSantos) [reply hawkmoth/util/doccompat.py:19]: Alright. looking forward to it then ;)

**Pull request merged**